packages/aws-cdk-lib/aws-ecr/lib/repository.ts

```diff
@@ -1045,6 +1045,12 @@ function validateAnyRuleLast(rules: LifecycleRule[]) {
  * Render the lifecycle rule to JSON
  */
 function renderLifecycleRule(rule: LifecycleRule) {
+  // If the rule is an instance of LifecycleRuleClass, use its toJSON method
```

Comment:
`LifecycleRuleClass` never implemented `LifecycleRule` (nor should it, because `LifecycleRule` is data, not a class interface) so this can never fire.